### PR TITLE
Added Momentum Optimizer to 'optimization' in machine-learning

### PR DESCRIPTION
**Pull request by showmyth**
# Pull Request Template

## Description

This PR adds the `Momentum optimizater`  to the optimization module.

Momentum accelerates gradient descent by accumulating past gradients into a velocity vector, leading to faster and more stable convergence.

**Update equations:**
- `velocity_{k+1} = beta * velocity_k + gradient(x_k)`
- `x_{k+1} = x_k - learning_rate * velocity_{k+1}`

For more information, checkout [Momentum Optimization by CS231](https://cs231n.github.io/neural-networks-3/#sgd)

## Type of change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I added my algorithm to the corresponding `mod.rs` file within its own folder, and in any parent folder(s).
- [x] I added my algorithm to `DIRECTORY.md` with the correct link.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

Please make sure that if there is a test that takes too long to run ( > 300ms), you `#[ignore]` that or
try to optimize your code or make the test easier to run. We have this rule because we have hundreds of
tests to run; If each one of them took 300ms, we would have to wait for a long time.

### CODE REVIEW

**Comment by showmyth** [comment]
Hey @siriak, would be grateful if you could review this as per your convinience.

**Review by siriak — APPROVED** [review]
Looks good, thanks!
